components/script/dom/dedicatedworkerglobalscope.rs

```diff
@@ -231,6 +232,7 @@ impl DedicatedWorkerGlobalScope {
 
             {
                 let _ar = AutoWorkerReset::new(&global, worker);
+                let _aes = AutoEntryScript::new(global.upcast());
```

Comment:
Instead of manual RAII, can't we just call that in `execute_script`? Seems like just before it is the only call site you changed, no?